### PR TITLE
Added toggle memory relay function to mlvl_wrapper.MlvlArea

### PR DESCRIPTION
**Pull request by UltiNaruto**

### CODE REVIEW

**Comment by UltiNaruto** [comment]
the diff in patches.rs that are not related to memory relays is when I convert all tabs to spaces and I press Alt + Shift + S in Notepad++ to remove any spaces in an empty line

File: src/mlvl_wrapper.rs [hunk 1]
```diff
@@ -63,6 +63,29 @@ impl<'r, 'mlvl, 'cursor, 'list> MlvlArea<'r, 'mlvl, 'cursor, 'list>
         x.kind.as_mrea_mut().unwrap()
     }
 
+    pub fn toggle_memory_relay(&mut self, mem_relay_id: u32, active: u8)
+    {
+        let layer_id = ((mem_relay_id >> 26) & 0x1f) as usize;
+
+        for mem_relay in self.memory_relay_conns.iter_mut() {
```
Comment (toasterparty) [inline src/mlvl_wrapper.rs:86]: Please panic if the memory relay is not found

File: src/mlvl_wrapper.rs [hunk 2]
```diff
@@ -63,6 +63,29 @@ impl<'r, 'mlvl, 'cursor, 'list> MlvlArea<'r, 'mlvl, 'cursor, 'list>
         x.kind.as_mrea_mut().unwrap()
     }
 
+    pub fn toggle_memory_relay(&mut self, mem_relay_id: u32, active: u8)
```
Comment (toasterparty) [inline src/mlvl_wrapper.rs]: Toggle is the wrong word. Your function is a setter for the default state of the relay.

**Review by toasterparty — CHANGES_REQUESTED** [review]
Did we try using memory relays instead of special function script layer controllers for map dots?

File: src/mlvl_wrapper.rs (hunk 2 above)
Comment (UltiNaruto) [reply src/mlvl_wrapper.rs]: set_memory_relay_active then

**Comment by UltiNaruto** [comment]
> Did we try using memory relays instead of special function script layer controllers for map dots?

The problem is removing the dot and marking the pickup as inactive is independent from showing up on the map

File: src/mlvl_wrapper.rs (hunk 1 above)
Comment (UltiNaruto) [reply src/mlvl_wrapper.rs:86]: Done

File: src/mlvl_wrapper.rs (hunk 2 above)
Comment (UltiNaruto) [reply src/mlvl_wrapper.rs]: Done

**Review by toasterparty — APPROVED** [review]

**Pull request merged**